tasmota/tasmota_xsns_sensor/xsns_114_amsx915.ino

```diff
@@ -0,0 +1,251 @@
+*/
+
+#ifdef USE_I2C
+#ifdef USE_AMSX915
+/*********************************************************************************************\
+ * AMS5915, AMS6915 - Pressure and Temperature
+ *
+ * Source: Bastian Urschel
+ *
+ * I2C Address: 0x28
+\*********************************************************************************************/
+
+#define XSNS_114            114
+#define XI2C_86              86  // See I2CDEVICES.md
+
+#ifndef AMSX915_ADDR
+  #define AMSX915_ADDR         0x28
+#endif
+
+#define PMIN_DEFAULT      0
+#define PMAX_DEFAULT      0
+
+#ifndef USE_UFILESYS
+#warning "AMSx915 pressure settings cannot be saved persistent due to missing filesystem"
+#endif
+
+/********************************************************************************************/
+
+struct amsx915_data {
+  float pressure;
+  float temperature;
+  bool present;
+} amsx915data;
+
+const uint16_t AMSX915_VERSION = 0x0100;       // Latest sensor version (See settings deltas below)
+
+// Global structure containing driver saved variables
+struct {
+  uint32_t  crc32;    // To detect file changes
+  uint16_t  version;  // To detect driver function changes
+  int16_t   pmin;
+  int16_t   pmax;
+} amsx915Settings;
+
+void AmsDetect(void) {
+  // sensor datasheets does not provide any information/commands on sensor
+  // specific register so it is not possible to properly detect sensor type
+  if(Wire.available() != 4) {
+    // Error
+    amsx915data.present = false;
+    AddLog(LOG_LEVEL_INFO, PSTR("CFG: AMS not detected!"));
+
+    return;
+  }
+  while(Wire.available()) {
+    Wire.read();
+  }
+  amsx915data.present = true;
+  I2cSetActiveFound(AMSX915_ADDR, "AMSx915");
+    AddLog(LOG_LEVEL_INFO, PSTR("CFG: AMS detected"));
+}
+
+bool AmsCommand() {
+  int32_t vals[2];
+  ParseParameters(2,(uint32_t *)vals);
+  if(XdrvMailbox.data_len >= 3 && XdrvMailbox.data_len < 13) {
+    if (vals[0] >= -32768 && vals[0] < 32768) {
+      if (vals[1] >= -32768 && vals[1] < 32768) {
+        amsx915Settings.pmin = (int16_t)vals[0]; // save pmin of sensor
+        amsx915Settings.pmax = (int16_t)vals[1];  // same with pmax
+        AmsSettingsSave();
+        Response_P(S_JSON_SENSOR_INDEX_SVALUE, XSNS_114, "pressure range set");
+        return true;
+      }
+    }
+    Response_P(S_JSON_SENSOR_INDEX_SVALUE, XSNS_114, "invalid pressure range [-32768..32767]");
+    return false;
+  }
+  return false;
+}
+
+void AmsReadData(void) {
+  uint8_t buffer[4];
+  Wire.requestFrom(AMSX915_ADDR, 4);
+  if(Wire.available() != 4) {
+    Serial.println("AMS error read!");
+    amsx915data.present = false;
+    return;
+  }
+  buffer[0] = Wire.read();
+  buffer[1] = Wire.read();
+  buffer[2] = Wire.read();
+  buffer[3] = Wire.read();
+
+  amsx915data.pressure = ((256*(buffer[0]&0x3F)+buffer[1])-1638.0)*(amsx915Settings.pmax-amsx915Settings.pmin)/13107+amsx915Settings.pmin;
+  amsx915data.temperature = (((256.0*buffer[2]+buffer[3])*200.0)/65536)-50;
```

Comment:
Same, force every literal to `float`, otherwise the computation is done with `double` and the result is converted to `float`. Which increases A LOT the code size